### PR TITLE
Prow sub can work with GitHub app

### PR DESCRIPTION
**Pull request by chaodaiG**

### CODE REVIEW

File: prow/cmd/sub/main.go [hunk 1]
```diff
@@ -107,6 +107,9 @@ func main() {
 	if flagOptions.github.TokenPath != "" {
 		tokens = append(tokens, flagOptions.github.TokenPath)
 	}
+	if flagOptions.github.AppPrivateKeyPath != "" {
```
Comment (alvaroaleman) [inline prow/cmd/sub/main.go:110]: This shouldnt be needed, it happens in the flagutil when you call `GitHubClientWithLogFields`: https://github.com/kubernetes/test-infra/blob/ef5dc9a4f6b4dc988050e83d438ac0484439070f/prow/flagutil/github.go#L272-L274


Below is another check for `flagOptions.github.TokenPath `, that one needs to also allow apps auth

Comment (chaodaiG) [reply prow/cmd/sub/main.go:110]: ah. right. that's strange then. I'll take another looks

Comment (chaodaiG) [reply prow/cmd/sub/main.go:110]: found it, fixing in https://github.com/kubernetes/test-infra/pull/24317, and by the way removing duplicated logic in the PR

**Pull request merged**